Makefile

```diff
@@ -968,14 +986,14 @@ $(TARGET_ELF):  $(TARGET_OBJS)
 # Compile
 $(OBJECT_DIR)/$(TARGET)/%.o: %.c
 	$(V1) mkdir -p $(dir $@)
-	$(V1) $(if $(findstring $(subst ./src/main/,,$<), $(SPEED_OPTIMISED_SRC)), \
+	$(V1) $(if $(findstring $(notdir $<),$(SPEED_OPTIMISED_SRC)), \
```

Comment:
Hmm, ok. I have seen we have different files with identical names, why is that? Is that not a problem in it self?? Will see if I can revert this.
One example of base file name clashes we have:

    $ find . -name system_stm32f10x.c -print
    ./src/main/drivers/system_stm32f10x.c
    ./lib/main/CMSIS/CM3/DeviceSupport/ST/STM32F10x/system_stm32f10x.c
Is this a good practise?